src/java_bytecode/expr2java.cpp

```diff
@@ -235,6 +237,63 @@ std::string expr2javat::convert_constant(
     dest+='L';
     return dest;
   }
+  else if((src.type()==java_float_type()) ||
+          (src.type()==java_double_type()))
+  {
+    ieee_floatt ieee_repr(to_constant_expr(src));
+    std::string result;
+
+    bool float_flag=(src.type()==java_float_type());
+    if(ieee_repr.is_zero())
+    {
+      result+='0';
+      format_spect fs;
+      if(fs.precision>0)
```

Comment:
This is a bit odd -- either factor out the default if that's what you intend, or initialise this from somewhere